### PR TITLE
Add np.diag_indices_from numpy function

### PR DESCRIPTION
**Pull request by ashridh**
Addresses: #70

### CODE REVIEW

**Comment by ayush-1506** [comment]
The build failure doesn't seem related to my changes.

File: jax/numpy/lax_numpy.py [hunk 1]
```diff
@@ -2573,6 +2573,10 @@ def diag_indices(n, ndim=2):
                      .format(ndim))
   return (lax.iota(int_, n),) * ndim
 
+@_wraps(np.diag_indices_from)
+def diag_indices_from(arr):
+  return diag_indices(arr.shape[0], ndim=arr.ndim)
```
Comment (jakevdp) [inline jax/numpy/lax_numpy.py:2578]: numpy returns an error if the array is not square:
```python
>>> import numpy as np
>>> np.diag_indices_from(jnp.zeros((5, 8)))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-1c7a827c990f> in <module>()
      1 import numpy as np
----> 2 np.diag_indices_from(jnp.zeros((5, 8)))

<__array_function__ internals> in diag_indices_from(*args, **kwargs)

/usr/local/lib/python3.6/dist-packages/numpy/lib/index_tricks.py in diag_indices_from(arr)
    980     # all dimensions equal, so we check first.
    981     if not alltrue(diff(arr.shape) == 0):
--> 982         raise ValueError("All dimensions of input must be of equal length")
    983 
    984     return diag_indices(arr.shape[0], arr.ndim)

ValueError: All dimensions of input must be of equal length
```
We should have a similar check here.

File: tests/lax_numpy_test.py [hunk 2]
```diff
@@ -1565,6 +1565,20 @@ def testDiagIndices(self, ndim, n):
...
+      {"testcase_name": "arr_shape={}".format(
+        jtu.format_shape_dtype_string(shape, dtype)
+      ),
+       "dtype": dtype, "shape": shape}
+      for dtype in default_dtypes
+      for shape in [(1,1), (2,2), (3,3), (4,4), (5,5)]))
+  def testDiagIndicesFrom(self, dtype, shape):
+    rng = jtu.rand_default(self.rng())
+    np_fun = lambda arg : np.diag_indices_from(arg)
+    jnp_fun = lambda arg : jnp.diag_indices_from(arg)
```
Comment (jakevdp) [inline tests/lax_numpy_test.py:1578]: The ``lambda`` wrapper is superfluous here: you can use ``np_fun = np.diag_indices_from`` and similar for ``jnp``.

File: tests/lax_numpy_test.py [hunk 3]
```diff
@@ -1565,6 +1565,20 @@ def testDiagIndices(self, ndim, n):
...
+      {"testcase_name": "arr_shape={}".format(
+        jtu.format_shape_dtype_string(shape, dtype)
+      ),
+       "dtype": dtype, "shape": shape}
+      for dtype in default_dtypes
+      for shape in [(1,1), (2,2), (3,3), (4,4), (5,5)]))
+  def testDiagIndicesFrom(self, dtype, shape):
+    rng = jtu.rand_default(self.rng())
+    np_fun = np.diag_indices_from(arg)
+    jnp_fun = jnp.diag_indices_from(arg)
```
Comment (jakevdp) [inline tests/lax_numpy_test.py:1578]: These need to be functions, not the result of functions. i.e.
```
np_fun = np.diag_indices_from
jnp_fun = jnp.diag_indices_from
```

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: tests/lax_numpy_test.py (hunk 3 above)
Comment (ayush-1506) [reply tests/lax_numpy_test.py:1578]: Ah, right. Thanks!

**Comment by ayush-1506** [comment]
@jakevdp Done, kindly let me know if there is something else I'm missing.

**Comment by jakevdp** [comment]
Looks good, thanks!

**Pull request merged**